[C++ API] torch::nn::GroupNorm and F::group_norm

yf225 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29933 [C++ API] torch::nn::FractionalMaxPool{2,3}d module and functional
* #29922 [C++ API] torch::nn::GLU and F::glu
* **#29920 [C++ API] torch::nn::GroupNorm and F::group_norm**

Differential Revision: [D18539314](https://our.internmc.facebook.com/intern/diff/D18539314)

Commit:
[C++ API] torch::nn::GroupNorm and F::group_norm

Commit:
Update on "[C++ API] torch::nn::GroupNorm and F::group_norm"

[C++ API] torch::nn::GroupNorm and F::group_norm

gh-metadata: pytorch pytorch 29920 gh/yf225/75/head

pbelevich commented on torch/csrc/api/include/torch/nn/modules/normalization.h:
```diff
@@ -99,5 +101,38 @@ class TORCH_API CrossMapLRN2dImpl : public torch::nn::Cloneable<CrossMapLRN2dImp
+  Tensor bias;
+};
+
+/// A `ModuleHolder` subclass for `GroupNorm`.
```
```suggestion
/// A `ModuleHolder` subclass for `GroupNormImpl`.
```

pbelevich commented on torch/csrc/api/include/torch/nn/modules/normalization.h:
```diff
@@ -99,5 +101,38 @@ class TORCH_API CrossMapLRN2dImpl : public torch::nn::Cloneable<CrossMapLRN2dImp
+};
+
+/// A `ModuleHolder` subclass for `GroupNorm`.
+/// See the documentation for `GroupNorm` class to learn what methods it
```
```suggestion
/// See the documentation for `GroupNormImpl` class to learn what methods it
```

Commit:
Update on "[C++ API] torch::nn::GroupNorm and F::group_norm"

[C++ API] torch::nn::GroupNorm and F::group_norm

gh-metadata: pytorch pytorch 29920 gh/yf225/75/head

pbelevich commented on test/cpp/api/modules.cpp:
```diff
@@ -2830,6 +2848,15 @@ TEST_F(ModulesTest, PrettyPrintLayerNorm) {
+  ASSERT_EQ(
+    c10::str(GroupNorm(GroupNormOptions(2, 2))),
+      "torch::nn::GroupNorm(2, 2, eps=1e-05, affine=true)");
+      ASSERT_EQ(
```
something happened with indent

Commit:
Update on "[C++ API] torch::nn::GroupNorm and F::group_norm"

[C++ API] torch::nn::GroupNorm and F::group_norm

gh-metadata: pytorch pytorch 29920 gh/yf225/75/head
